pybleau/app/model/plot_template_manager.py

```diff
@@ -0,0 +1,49 @@
+import os
+from pathlib import Path
+
+from traits.api import cached_property, Str, List, Instance, Directory, \
+    Property, Event, HasStrictTraits
+
+from pybleau.app.plotting.i_plot_template_interactor import \
+    IPlotTemplateInteractor
+
+
+class PlotTemplateManager(HasStrictTraits):
+    """ Model for the dialog used to manage plot templates
+    """
+    #: Names of all the available plot templates
+    names = Property(List(Str), depends_on="templates_changed")
+
+    #: IPlotTemplateInteractor used to save/load/locate plot templates
+    interactor = Instance(IPlotTemplateInteractor, args=())
+
+    #: Directory of plot templates
+    plot_template_directory = Property(Directory)
+
+    #: Set to True whenever the list of templates changes or to refresh list
+    templates_changed = Event
+
+    # Public interface --------------------------------------------------------
+
+    def delete_templates(self, templates_names: list):
+        removed = self.interactor.delete_templates(templates_names)
+        if removed:
+            self.templates_changed = True
+
+    def rescan_template_dir(self):
+        self.templates_changed = True
```

Comment:
Caching the `names` property is essential for (at least my understanding) of how to make the `names` work. Without caching it, the list of names is updated arbitrarily, and brings in any new names that get added in an unpredictable way. Happy to discuss more, if we can find other ideas.

Reply:
Yes, let's discuss. Current implementation works but is a bit mysterious.

Reply:
We discussed offline and agreed on a newer implementation where a function is called instead of using these events.

> Yes, let's discuss. Current implementation works but is a bit mysterious.